### PR TITLE
[2/2] Adjust visual id migration tags

### PR DESCRIPTION
**Pull request by mahdibenromdhane**

### CODE REVIEW

**Comment by jerome-benoit** [comment]
@mahdibenromdhane That version is bug free?

**Comment by mahdibenromdhane** [comment]
> @mahdibenromdhane That version is bug free?

Yes.

File: src/storage/mongodb/MongoDBStorage.ts [hunk 1]
```diff
@@ -105,7 +105,6 @@ export default class MongoDBStorage {
       { fields: { deleted: 1, createdOn: 1 } },
       { fields: { issuer: 1, createdOn: 1 } },
       { fields: { userID: 1, issuer: 1 } },
-      { fields: { visualID: 1 }, options: { unique: true } }
```
Comment (jerome-benoit) [inline src/storage/mongodb/MongoDBStorage.ts:108]: The goal is to have an index on that new field and to ensure its uniqueness, right? So you probably have removed that line by mistake.

**Review by jerome-benoit — CHANGES_REQUESTED** [review]

File: src/migration/tasks/AddVisualIDPropertyToTagsTask.ts [hunk 2]
```diff
@@ -19,19 +20,23 @@ export default class AddVisualIDPropertyToTagsTask extends MigrationTask {
...
+            { $set: { visualID: visualID } });
+          updated++;
+        }
       }
-    }
+      skip += Constants.BATCH_PAGE_SIZE;
+    } while (skip < count);
     // Log in the default tenant
     if (updated > 0) {
       await Logging.logDebug({
```
Comment (jerome-benoit) [inline src/migration/tasks/AddVisualIDPropertyToTagsTask.ts:42]: The migration version is not bumped.

**Review by jerome-benoit — CHANGES_REQUESTED** [review]

File: src/storage/mongodb/MongoDBStorage.ts (hunk 1 above)
Comment (mahdibenromdhane) [reply src/storage/mongodb/MongoDBStorage.ts:108]: The unique index field need to be added after filling the visualID so we need to merge in 3 steps:

1. Remove unique index
2. fill the visualID
3. Re Add the index

**Review by jerome-benoit — APPROVED** [review]

File: src/migration/tasks/AddVisualIDPropertyToTagsTask.ts (hunk 2 above)
Comment (mahdibenromdhane) [reply src/migration/tasks/AddVisualIDPropertyToTagsTask.ts:42]: Fixed

**Pull request merged**